cmd/scripts_cmd.go

```diff
@@ -164,3 +164,60 @@ func ScriptDelete(c *cli.Context) error {
 	}
 	return nil
 }
+
+// ScriptAttachmentAdd subcommand function
+func ScriptAttachmentAdd(c *cli.Context) error {
+	debugCmdFuncInfo(c)
+	scriptSvc, formatter := WireUpScript(c)
+
+	checkRequiredFlags(c, []string{"id", "name", "filepath"}, formatter)
+
+	sourceFilePath := c.String("filepath")
+	if !utils.FileExists(sourceFilePath) {
+		formatter.PrintFatal("Invalid file path", fmt.Errorf("no such file or directory: %s", sourceFilePath))
+	}
+
+	attachmentIn := map[string]interface{}{
+		"name": c.String("name"),
+	}
+
+	// adds new attachment
+	attachment, err := scriptSvc.AddScriptAttachment(&attachmentIn, c.String("id"))
+	if err != nil {
+		formatter.PrintFatal("Couldn't add attachment to script", err)
+	}
+
+	// uploads new attachment file
+	err = scriptSvc.UploadScriptAttachment(sourceFilePath, attachment.UploadURL)
+	if err != nil {
+		formatter.PrintFatal("Couldn't upload attachment data", err)
+	}
```

Comment:
If there was an error with the Upload or the Uploaded it would be nice that the CLI tried to delete the attachment to clean up. That may apply as well to cookbook versions.

Reply:
Done, implemented as well for cookbook versions